pkg/list/list_stacks.go

```diff
@@ -1,45 +1,386 @@
+				}
 			}
-			if _, exists := terraform[component]; exists {
-				filteredStacks = append(filteredStacks, stackName)
+		}
+
+		// Extract stage from stack name if not set in vars
+		if _, ok := stackInfo["vars"].(map[string]any)["stage"]; !ok {
+			// Only set stage from stack name if it's not already set in vars
+			if stackName != stackInfo["atmos_stack_file"].(string) {
+				stackInfo["vars"].(map[string]any)["stage"] = stackName
+			}
+		}
+
+		// Copy other stack configuration
+		for k, v := range v2 {
+			if k != "vars" && k != "components" {
+				stackInfo[k] = v
+			}
+		}
+
+		return stackInfo
+	}
+
+	// Helper function to check if stack has component
+	hasComponent := func(components map[string]any, targetComponent string) bool {
+		for _, section := range components {
+			if compSection, ok := section.(map[string]any); ok {
+				if _, exists := compSection[targetComponent]; exists {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	var filteredStacks []map[string]any
+
+	// Filter and process stacks
+	for stackName, stackData := range stacksMap {
+		v2, ok := stackData.(map[string]any)
+		if !ok {
+			continue
+		}
+
+		if component != "" {
+			// Only include stacks with the specified component
+			if components, ok := v2["components"].(map[string]any); ok && hasComponent(components, component) {
+				stackInfo := createStackInfo(stackName, v2)
+				filteredStacks = append(filteredStacks, stackInfo)
 			}
+		} else {
+			// Include all stacks when no component filter is specified
+			stackInfo := createStackInfo(stackName, v2)
+			filteredStacks = append(filteredStacks, stackInfo)
 		}
+	}
 
-		if len(filteredStacks) == 0 {
+	if len(filteredStacks) == 0 {
+		if component != "" {
 			return fmt.Sprintf("No stacks found for component '%s'"+"\n", component), nil
 		}
-		sort.Strings(filteredStacks)
-		return strings.Join(filteredStacks, "\n") + "\n", nil
+		return "No stacks found\n", nil
+	}
+
+	// Sort stacks by name
+	sort.Slice(filteredStacks, func(i, j int) bool {
+		return filteredStacks[i]["atmos_stack"].(string) < filteredStacks[j]["atmos_stack"].(string)
+	})
+
+	// If no columns are configured, use default columns
+	if len(listConfig.Columns) == 0 {
+		// Define all possible columns
+		allColumns := []schema.ListColumnConfig{
+			{Name: "Stack", Value: "{{ .atmos_stack }}"},
+			{Name: "Tenant", Value: "{{ getVar .vars \"tenant\" }}"},
+			{Name: "Environment", Value: "{{ getVar .vars \"environment\" }}"},
+			{Name: "Stage", Value: "{{ getVar .vars \"stage\" }}"},
+			{Name: "File", Value: "{{ .atmos_stack_file }}"},
+		}
+
+		// Helper function to check if a column has any non-empty values
+		hasValues := func(col schema.ListColumnConfig) bool {
+			// Stack and File columns are always shown
+			if col.Name == "Stack" || col.Name == "File" {
+				return true
+			}
+
+			funcMap := template.FuncMap{
+				"getVar": func(vars map[string]any, key string) string {
+					if val, ok := vars[key]; ok && val != nil {
```

Comment:
Makes me think we are restricted to only accessing vars, when the entire object of a component in a stack should be accessible.